docs/DafnyRef/Expressions.md

```diff
@@ -1022,6 +1022,48 @@ function GhostF(z: Stuff): int
 }
 ```
 
+## Let or Fail Expression
+````grammar
+LetExpr(allowLemma, allowLambda) =
+  [ "ghost" ] "var" CasePattern { "," CasePattern }
+  ( ":-" | { Attribute } ":|" )
```

Comment:
```suggestion
  { Attribute } ":|"
```

Reply:
This is not the fix, but fixed up grammar.